aws_lambda_builders/workflows/nodejs_npm/utils.py

```diff
@@ -48,3 +66,189 @@ def abspath(self, path):
+            NPM process
+
+        Returns
+        -------
+        string
+            Path to the packed tar file
+        """
+        manifest_path = osutils.normpath(manifest_path)
+        LOG.debug("NODEJS processing %s", manifest_path)
+        manifest_hash = str(abs(hash(manifest_path)))
+        if manifest_hash in packed_manifests:
+            # Already processed or circular dependency
+            # If empty, it will be removed from the manifest
+            return packed_manifests[manifest_hash]
+        packed_manifests[manifest_hash] = ""
+        manifest_dir = osutils.dirname(manifest_path)
+        manifest_scratch_dir = osutils.joinpath(scratch_dir, manifest_hash)
+        manifest_scratch_package_dir = osutils.joinpath(manifest_scratch_dir, "package")
+        osutils.makedirs(manifest_scratch_package_dir)
+
+        # Pack and copy module to scratch so that we don't update the customers files in place
+        DependencyUtils.ship_module(manifest_dir, scratch_dir, manifest_scratch_dir, osutils, subprocess_npm)
+
+        # Process local children dependencies
+        local_dependencies = DependencyUtils.get_local_dependencies(manifest_path, osutils)
+
+        for (dep_name, dep_path) in local_dependencies.items():
+            if dep_path.startswith("file:"):
+                dep_path = dep_path[5:].strip()
+            dep_manifest = osutils.joinpath(manifest_dir, dep_path, "package.json")
+
+            tar_path = DependencyUtils.package_dependencies(
+                dep_manifest, scratch_dir, packed_manifests, osutils, subprocess_npm
+            )
+
+            manifest_scratch_path = osutils.joinpath(manifest_scratch_package_dir, "package.json")
+
+            # Make a backup we will use to restore in the final build folder
+            osutils.copy_file(manifest_scratch_path, manifest_scratch_path + ".bak")
+
+            DependencyUtils.update_manifest(manifest_scratch_path, dep_name, tar_path, osutils)
+
+        # Pack the current dependency
+        tarfile_name = subprocess_npm.run(["pack", "-q", manifest_scratch_package_dir], cwd=scratch_dir).splitlines()[
+            -1
+        ]
+
+        packed_manifests[manifest_hash] = osutils.joinpath(scratch_dir, tarfile_name)
+
+        LOG.debug("NODEJS %s packed to %s", manifest_scratch_package_dir, packed_manifests[manifest_hash])
+
+        return packed_manifests[manifest_hash]
+
+    @staticmethod
+    def update_manifest(manifest_path, dep_name, dependency_tarfile_path, osutils):
+        """
+        Helper function to update dependency path to localized tar
+
+        Parameters
+        ----------
+        manifest_path : str
+            Manifest path to update
+        dep_name : str
+            Dependency name
+        dependency_tarfile_path : str
+            Packed dependency tar file path
+        osutils : OSUtils
+            OS utils
+        """
+        with osutils.open_file(manifest_path, "r") as manifest_read_file:
```

Comment:
This will break on Windows with UTF-8 bytes as it'll fall back to default encoding (usually Latin-1 on Windows). `open_file` can likely default to UTF-8 instead.

Reply:
Good catch, we have to update the method to pass `encoding="utf-8"` to `open` then.